template.yaml

```diff
@@ -1,27 +1,28 @@
 AWSTemplateFormatVersion: '2010-09-09'
 Transform: AWS::Serverless-2016-10-31
 Description: >
-  Hello World lambda
+  Send SNS message to Synapse user as email
 
 # More info about Globals: https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst
 Globals:
   Function:
     Timeout: 3
 
 Resources:
-  HelloWorldFunction:
+  SendBudgetAlertFunction:
     Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
     Properties:
-      CodeUri: hello_world/
+      CodeUri: send_budget_alert/
```

Comment:
There are 2 options to access secrets, first one is the one I referenced above the 2nd one is to use a role to get direct access to the SSM.   @tthyer and i agree that doing it with a role is more secure and better way to go.  Since @tthyer has already setup the foundation for doing it this way it will not require much work to use that pattern.   You can take a look at [cfn-cr-alb-rule](https://github.com/Sage-Bionetworks-IT/cfn-cr-alb-rule) to see the pattern for your lambda to retrieve secrets

Reply:
I implemented the ssm approach in the latest commit:  https://github.com/Sage-Bionetworks-IT/lambda-send-budget-alert/pull/1/commits/4b1ca9b6f5c942a09dec09c5360d7417f04b8955
This introduces two input parameters to the CFN template, for the names of the two SSM secrets.